tuf/client/client.go

```diff
@@ -111,10 +110,6 @@ func (c Client) checkRoot() error {
 		return fmt.Errorf("Cached root sha256 did not match snapshot root sha256")
```

Comment:
Isn't it possible that another byte array could still collide with the same hash?  Especially if we are not restricting the length?  It is improbable, but the size check below make it even more so.

Also, while `c.cache.GetMeta` takes a byte `size` to read, it doesn't actually restrict the size in the `filestore.go` implementation.  I think this gives us more reason to leave in the sanity check for the length below.